txnsync/txngroups.go

```diff
@@ -44,11 +44,50 @@ func encodeTransactionGroups(inTxnGroups []transactions.SignedTxGroup) []byte {
 	return stub.MarshalMsg(protocol.GetEncodingBuf()[:0])
 }
 
-func decodeTransactionGroups(bytes []byte) (txnGroups []transactions.SignedTxGroup, err error) {
+func encodeTransactionGroups(inTxnGroups []transactions.SignedTxGroup) ([]byte, error) {
+	txnCount := 0
+	for _, txGroup := range inTxnGroups {
+		txnCount += len(txGroup.Transactions)
+	}
+	stub := txGroupsEncodingStub{
+		TotalTransactionsCount: uint64(txnCount),
+		TransactionGroupCount:  uint64(len(inTxnGroups)),
+		TransactionGroupSizes:  make([]byte, 0, len(inTxnGroups)),
+	}
+
+	index := 0
+	for _, txGroup := range inTxnGroups {
+		if len(txGroup.Transactions) > 1 {
+			for _, txn := range txGroup.Transactions {
+				if err := stub.deconstructSignedTransactions(index, txn); err != nil {
+					return nil, fmt.Errorf("failed to encodeTransactionGroups: %v", err)
+				}
+				index++
+			}
+			stub.TransactionGroupSizes = append(stub.TransactionGroupSizes, byte(len(txGroup.Transactions)-1))
+		}
+	}
+	stub.TransactionGroupSizes = squeezeByteArray(stub.TransactionGroupSizes)
+	for _, txGroup := range inTxnGroups {
+		if len(txGroup.Transactions) == 1 {
+			for _, txn := range txGroup.Transactions {
+				if err := stub.deconstructSignedTransactions(index, txn); err != nil {
+					return nil, fmt.Errorf("failed to encodeTransactionGroups: %v", err)
+				}
+				index++
+			}
+		}
+	}
+	stub.finishDeconstructSignedTransactions()
+
+	return stub.MarshalMsg(protocol.GetEncodingBuf()[:0]), nil
+}
+
+func decodeTransactionGroupsOld(bytes []byte) (txnGroups []transactions.SignedTxGroup, err error) {
```

Comment:
decodeTransactionGroupsOld along with txGroupsEncodingStubOld seems to be redundant.

Reply:
I left this in the code to compare performance, should I go ahead and delete all of these?

Reply:
I think that it's good to leave it as a reference; in that case:
1. document it, so we'll know that it's not being used.
2. move this method to the XXX_test.go file.

Reply:
Okay, I moved the decode / encode to the test file but left the data structures since msgp wouldnt compile them and left a comment